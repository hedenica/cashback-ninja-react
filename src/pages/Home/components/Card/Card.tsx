import { formatCurrency, getCashbackByOrder, Product } from 'resources'

import shoppingBag from 'assets/shopping-bag.svg'
import cashbackImg from 'assets/cashback.svg'

import styles from './card.module.scss'

interface Orders {
  id: number;
  products: Product[];
  total: number;
}

interface CardProps {
  orders: Orders[];
}

export function Card ({ orders }: CardProps) {
  return (
    <div className={styles.cardColumns}>
      {orders.map(order => (
        <div key={order.id} className={styles.card}>
          <div className={styles.banner}>
            <div className={styles.order}>
              <img src={shoppingBag} alt="sacola de compras" />
              <p>Pedido | NÂº {order.id}</p>
            </div>
          </div>
          <div className={styles.itensContainer}>
            <div className={styles.itensList}>
              {order.products.map((product, index) => (
                <p key={index}>{product.name}</p>
              ))}
            </div>

            <div className={styles.itensPrice}>
              {order.products.map((product, index) => (
                <p key={index}>{formatCurrency(product.price)}</p>
              ))}
            </div>
          </div>
          <div className={styles.cardCashback}>
            <img src={cashbackImg} alt="porquinho" />
            <span>
              cashback | {formatCurrency(getCashbackByOrder(order.total))}
            </span>
          </div>
        </div>
      ))}
    </div>
  )
}
