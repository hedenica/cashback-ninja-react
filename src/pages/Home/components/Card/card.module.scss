.cardColumns {
  display: grid;
  grid-template-columns: repeat(4, 2fr);
  row-gap: 2rem;
  column-gap: 2rem;
  align-items: center;
  justify-items: center;

  padding: 50px;
}

.card {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  width: 300px;

  height: 400px;
  min-height: 400px;

  border-radius: 12px;
  background: var(--white);
  box-shadow: 0px 1rem 1.5rem rgb(0 0 0 / 30%);

  transition: transform 0.2s;
}

.card:hover {
  transform: translate3D(0,-1px,0) scale(1.03);
}

.banner {
  background-image: url('../../../../assets/piggy.jpeg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: bottom;

  position: absolute;
  top: 0;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  left: 0;
  right: 0;
  
  height: 7rem;

  display: flex;
  align-items: flex-end;
  justify-content: center;

  box-sizing: border-box;
}

.order {
  width: 60%;
  height: 42px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 12px;
  background: var(--white);
  transform: translateY(50%);
  box-shadow: 0 0.5rem 1rem rgb(0 0 0 / 20%);

  img {
    width: 20px;
    height: 20px;
  }

  p {
    font-size: 1rem;
  }
}

.itensContainer {
  display: flex;
  justify-content: space-around;
  margin: 40px 0;
  align-items: flex-start;
  line-height: 1.8rem;
  font-size: 1rem;

  height: 200px;

  position: absolute;
  top: 112px;
  left: 0;
  right: 0;

  overflow: auto;
}

.cardCashback {
  background: var(--yellow);
  height: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
  border-bottom-left-radius: 12px;
  border-bottom-right-radius: 12px;
  font-weight: bold;
  letter-spacing: 0.1rem;

  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;

  img {
    margin-right: 8px;
  }
}
