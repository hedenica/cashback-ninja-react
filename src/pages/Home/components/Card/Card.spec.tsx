import { render } from '@testing-library/react'
import { Card } from './Card'

const MOCKED_ORDERS = [
  {
    id: 1,
    products: [{
      name: 'foo',
      price: 123,
    }],
    total: 123,
  },
]

describe('Card', () => {
  it('should render properly', () => {
    const { container } = render(<Card orders={MOCKED_ORDERS} />)

    expect(container).toMatchSnapshot()
  })
})
