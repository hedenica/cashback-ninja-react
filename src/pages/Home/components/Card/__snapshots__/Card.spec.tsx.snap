// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render properly 1`] = `
<div>
  <div
    class="cardColumns"
  >
    <div
      class="card"
    >
      <div
        class="banner"
      >
        <div
          class="order"
        >
          <img
            alt="sacola de compras"
            src="shopping-bag.svg"
          />
          <p>
            Pedido | Nº 
            1
          </p>
        </div>
      </div>
      <div
        class="itensContainer"
      >
        <div
          class="itensList"
        >
          <p>
            foo
          </p>
        </div>
        <div
          class="itensPrice"
        >
          <p>
            R$ 123,00
          </p>
        </div>
      </div>
      <div
        class="cardCashback"
      >
        <img
          alt="porquinho"
          src="cashback.svg"
        />
        <span>
          cashback | 
          R$ 12,30
        </span>
      </div>
    </div>
  </div>
</div>
`;
